contracts/Colony.sol

```diff
@@ -93,6 +94,33 @@ contract Colony is ColonyStorage, PatriciaTreeProofs {
     return ColonyAuthority(address(authority)).hasUserRole(_user, _domainId, uint8(_role));
   }
 
+  function hasInheritedUserRole(
+    address _user,
+    uint256 _domainId,
+    ColonyRole _role,
+    uint256 _childSkillIndex,
+    uint256 _childDomainId
+  ) public view returns (bool)
+  {
+    return (
+      hasUserRole(_user, _domainId, _role) &&
+      (_domainId == _childDomainId || validateDomainInheritance(_domainId, _childSkillIndex, _childDomainId))
+    );
+  }
+
+  function userCanSetRoles(
+    address _user,
+    uint256 _domainId,
+    uint256 _childSkillIndex,
+    uint256 _childDomainId
+  ) public view returns (bool)
+  {
+    return (
+      hasUserRole(_user, 1, ColonyRole.Root) ||
+      (_domainId != _childDomainId && hasInheritedUserRole(_user, _domainId, ColonyRole.Architecture, _childSkillIndex, _childDomainId))
```

Comment:
Mm I'm partial to treating ArchitectureSubdomain as a very low-level implementation detail and really never referring to it separately from Architecture -- it can't be assigned separately and so I don't really like the idea of referring to both Architecture and Subdomain in external calls.

Reply:
Why do we even have it then? Why did we not just have `_domainId != _childDomainId` in functions that are currently limited to `ArchitectureSubdomain`?

Reply:
Mm that's a fair question. At the time it seemed like the only way to implement the subdomain-only behavior but perhaps that was an oversight. Let me think about it some more.